### ISSUE TITLE
Improve analyzation time with multi threading

### ISSUE THREAD

**Issue by joglr** [git-truck/git-truck#612]
https://nodejs.org/api/worker_threads.html

**Comment by dawidwoz** [comment]
My ultimate goal is to create a process in the background where a user can browse the repo while we are working on the process of gathering the metrics.

There is a short summary of my findings for this issue:

1.  I've tried one worker for the hydrate phase - it works but, in the case, of the big repository it has the same problem as the solution without a worker. I often end up with this error:
![image](https://user-images.githubusercontent.com/52498300/204208908-2188c19d-2e8a-406e-8df0-ff843c5cd378.png)

It can be found here: [https://github.com/dawidwoz/git-truck/tree/dawidwoz/workers-one] (https://github.com/dawidwoz/git-truck/tree/dawidwoz/workers-one)

2.  I've tried many workers for the hydrate phase - it works but, in the case, of the big repository there are simply too many workers and I couldn't find a way to limit them while other tasks are waiting. The error is the same as above. 

3.  I've managed to load the repo without the hydrate phase, and then I wanted to do the call to git log on clicking the element using `UseEffect` or `loader`.  Here, the problem was the usage of `spawn` as `child_process` resulted in the error: `No matching export in "node-modules-polyfills:child_process" for import "spawn"`. I couldn't find a way around it.

Currently, I'm thinking about where it can be possible to use workers as my three ideas failed. 

cc: @mircealungu

**Comment by joglr** [comment]
A bit late to reply, but I also just experimented a bit with workers. I found the amount of overhead from using the workers, made the performance worse than just using a single thread. I tried by using the [workerpool](https://www.npmjs.com/package/workerpool) package running on a PC with 8 cores, 7 of them in the pool. 
![image](https://user-images.githubusercontent.com/1959615/216982718-f0e785b3-dbfa-43f7-88d4-1d8e00c2c6b6.png)

for `sqlite/sqlite`, the analyzation time is 264 seconds compared to 47.35 seconds with the current implementation. 

I still think doing some sort of progressive enhancement would be cool, to instantly show a structural overview of the system and then a progressbar with the analyzation.
